Framework/API/inc/MantidAPI/ExperimentInfo.h

```diff
@@ -74,11 +76,13 @@ class MANTID_API_DLL ExperimentInfo {
   void swapInstrumentParameters(Geometry::ParameterMap &pmap);
 
   /// Cache a lookup of grouped detIDs to member IDs
-  void cacheDetectorGroupings(const det2group_map &mapping);
-  /// Returns the detector IDs that make up the group that this ID is part of
-  const std::set<detid_t> &getGroupMembers(const detid_t detID) const;
-  /// Get a detector or detector group from an ID
-  Geometry::IDetector_const_sptr getDetectorByID(const detid_t detID) const;
+  virtual void cacheDetectorGroupings(const det2group_map &mapping);
+
+  void setNumberOfDetectorGroups(const size_t count);
+  void updateCachedDetectorGrouping(const size_t index,
+                                    const std::set<detid_t> &detIDs);
+  void addDetectorToGroup(const size_t index, const detid_t detID);
+  void addDetectorsToGroup(const size_t index, const std::set<detid_t> &detIDs);
```

Comment:
This is slightly off on a tangent, but have you considered what the addition of these methods might mean to the `MultipleExperimentInfos` interface as used by the `MDWorkspace` types? I would expect that the `ExperimentInfos` as seen by `MDWorkspaces` are essentially immutable. I'm not sure that its worth worrrying about that though at this stage.